src/drive/mobile/modules/upload/index.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react'
+import localforage from 'localforage'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { withRouter } from 'react-router'
+import PropTypes from 'prop-types'
+
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import { Query, cancelable, withClient } from 'cozy-client'
+
+import { translate } from 'cozy-ui/transpiled/react'
+import { Modal } from 'cozy-ui/transpiled/react'
+
+import { uploadFilesFromNative } from 'drive/web/modules/upload'
+
+import { ROOT_DIR_ID, TRASH_DIR_ID } from 'drive/constants/config'
+import Header from 'drive/web/modules/move/Header'
+import Explorer from 'drive/web/modules/move/Explorer'
+import FileList from 'drive/web/modules/move/FileList'
+import Loader from 'drive/web/modules/move/Loader'
+import LoadMore from 'drive/web/modules/move/LoadMore'
+import Footer from 'drive/web/modules/move/Footer'
+import Topbar from 'drive/web/modules/move/Topbar'
+
+export const generateForQueue = files => {
+  const filesForQueue = []
+  files.map(file => {
+    filesForQueue.push({ file: file, isDirectory: false })
+  })
+  return filesForQueue
+}
+
+export class DumbUpload extends Component {
+  state = {
+    items: null,
+    folderId: ROOT_DIR_ID
+  }
+  async componentDidMount() {
+    const items = await localforage.getItem('importedFiles')
+    this.setState({ items })
+  }
+
+  async uploadAndState() {
+    const { items, folderId } = this.state
+    const { router, uploadFilesFromNative } = this.props
+    const filesForQueue = generateForQueue(items)
+    uploadFilesFromNative(filesForQueue, folderId, this.callbackSuccess)
+    setTimeout(() => router.push(`/folder/${folderId}`), 500)
+  }
+  callbackSuccess = () => {
+    const { items } = this.state
+    const { t } = this.props
+    Alerter.success(t('ImportToDrive.success', { smart_count: items.length }))
+    localforage.removeItem('importedFiles')
+  }
+  navigateTo = folder => {
+    this.setState({ folderId: folder.id })
+  }
+
+  registerCancelable = promise => {
+    if (!this.promises) this.promises = []
+    const cancelableP = cancelable(promise)
+    this.promises.push(cancelableP)
+    return cancelableP
+  }
+
+  contentQuery = client => {
+    return client
+      .find('io.cozy.files')
+      .where({
+        dir_id: this.state.folderId,
+        _id: {
+          $ne: TRASH_DIR_ID
+        }
+      })
+      .indexFields(['dir_id', 'type', 'name', '_id'])
+      .sortBy([{ dir_id: 'asc' }, { type: 'asc' }, { name: 'asc' }])
+  }
+  breadcrumbQuery = client => {
+    return client.get('io.cozy.files', this.state.folderId)
+  }
+  onClose = () => {
+    const { router } = this.props
+    router.push('/')
+  }
+
+  render() {
+    const { items, folderId, uploadInProgress } = this.state
+    const { t } = this.props
+    if (items === null) return null
```

Comment:
Wouldn't it make more sense to start rendering the rest of the modal, even while the `items` aren't ready? That way we can start the queries earlier. And we can delay showing the file list until the items are ready for example.

Reply:
Good point!